File: core/src/main/java/pcb/uwu/units/derived/mechanics/AccelerationUnit.java
package pcb.uwu.units.derived.mechanics;

import pcb.uwu.core.CompositeUnit;
import pcb.uwu.core.UnitCounter;
import pcb.uwu.units.base.LengthUnit;
import pcb.uwu.units.base.TimeUnit;

public class AccelerationUnit extends CompositeUnit {

	// region base constructor

	public AccelerationUnit(LengthUnit lengthUnit, TimeUnit timeUnit) {
		super(new UnitCounter()
				.major(lengthUnit)
				.minor(timeUnit, 2));
	}

	// endregion

	// region derived constructors

	public AccelerationUnit(SpeedUnit lengthUnit, TimeUnit timeUnit) {
		super(new UnitCounter()
				.major(lengthUnit.getUnitCounter())
				.minor(timeUnit));
	}

	// endregion

}

File: core/src/main/java/pcb/uwu/unit/derived/mechanics/AccelerationUnit.kt
package pcb.uwu.unit.derived.mechanics

import pcb.uwu.core.CompositeUnit
import pcb.uwu.core.UnitCounter
import pcb.uwu.unit.base.LengthUnit
import pcb.uwu.unit.base.TimeUnit

class AccelerationUnit : CompositeUnit
{
    constructor(lengthUnit: LengthUnit,
                timeUnit: TimeUnit)
            : super(UnitCounter()
                        .major(lengthUnit)
                        .minor(timeUnit)
                        .minor(timeUnit))

    constructor(lengthUnit: SpeedUnit,
                timeUnit: TimeUnit)
            : super(UnitCounter()
                        .major(lengthUnit)
                        .minor(timeUnit))
}